app/Domain/User/Repository/UserCreatorRepository.php

```diff
@@ -1,9 +1,9 @@
 <?php
 namespace App\Domain\User\Repository;
 
-use App\Domain\User\DTO\CreateUserDTO;
+use App\Domain\User\User;
 
 interface UserCreatorRepository
```

Comment:
Não faz sentido pra min, você usar interface, e o nome do arquivo se chamar UserCreatorRepository.
Talvez teria mais sentido se chama-se de UserCreatorContract, e esta dentro de uma pasta de Contracts dentro da pasta repository.

Ou to falando besteira? 😄

Reply:
Oi Junior, então to usando inversão de depedência, por isso segui o nome com Repository, alvez neste caso mudaria para UserRepository ao inves deste nome que coloquei. 
O `UserCreatorRepository` ele chama o meu repositório Doctrine que tem essa interface implementada

Reply:
Pois é neste sentido sendo ele um Contrato, não faz sentido se chamar UserCreatorRepository, por que você bate e lê o nome do arquivo, opa aqui é um repositório, mas quando você abre o arquivo não é, é uma interface. :smile: 
Será que faz sentido só pra min ?

Reply:
Então, na vdd ele é o contrato que meu dominio deve seguir quando se trata de repositorio. 
Meu dominio não sabe o que implementei na Infra (DB), ele apenas segue o contrato. 
Se eu mudar o Doctrine pra PDO por exemplo, não vou precisar alterar no meu dominio (User) pois ele só acredita que o repositorio dele é a interface.

Reply:
Show, obrigado pela discussão Junior!